### PR TITLE
Tracking position on map

### PR DESCRIPTION
**Pull request by ikurek**

### CODE REVIEW

File: app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    SideEffect {
+        coroutineScope.launch {
+            when {
+                /**
+                 * THis handles the initial state of map, when it's not zoomed to anything,
+                 * and not already zooming, it should zoom to the trackable location,
+                 * before we start animating between the provided camera positions
+                 */
+                !mapState.isZoomedInToTrackable && mapState.location != null && !cameraPositionState.isMoving -> {
```
Comment (davyskiba) [inline app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt:46]: Could this be extracted to a function like `isInInitialState`?

Comment (davyskiba) [inline app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt:46]: Could this be extracted to a function like `isInInitialState`?

File: app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    animateMapToLocation(
+                        cameraPositionState,
+                        mapState.locationLatLng()!!,
+                        longAnimationDuration
+                    )
+                    viewModel.onZoomedToTrackablePosition()
+                }
+                /**
+                 * If it's already zoomed to the package, it should use the camera positions
```
Comment (davyskiba) [inline app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt:55]: package is a bit of surprise here 😅  replace with trackable for consistency?

Comment (davyskiba) [inline app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt:55]: package is a bit of surprise here 😅  replace with trackable for consistency?

**Review by davyskiba — CHANGES_REQUESTED** [review]
Please add instructions for google maps key creation

File: app/build.gradle [hunk 3]
```diff
@@ -18,6 +18,9 @@ android {
         vectorDrawables {
             useSupportLibrary true
         }
+
+        def GOOGLE_MAPS_API_KEY = localProperties.getProperty("GOOGLE_MAPS_API_KEY")
```
[the remark this answers is not in the document]
Comment (ikurek) [reply app/build.gradle:22]: Nice catch 😄 https://github.com/ably/asset-tracking-android-customer-app-demo/pull/4/commits/a48169595b31b1d7274055259fc6ce41682d1c25

File: app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt (hunk 1 above)
Comment (ikurek) [reply app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt]: I've moved it to `mapState` in https://github.com/ably/asset-tracking-android-customer-app-demo/pull/4/commits/0eb869d508157fe3024164f2a30ed4c6572cca19

File: app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt (hunk 2 above)
Comment (ikurek) [reply app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt]: Renamed in https://github.com/ably/asset-tracking-android-customer-app-demo/pull/4/commits/3f91b5b7301b2899ed5a6b11949442fe1a6f7efe

**Review by davyskiba — APPROVED** [review]

**Pull request merged**